domains/Discussions/Tests/Feature/QuestionsIndexTest.php

```diff
@@ -0,0 +1,235 @@
+
+        QuestionFactory::new([
+            'author_id' => $user->id,
+        ])
+            ->count(3)
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'author' => $user->id,
+        ]))
+            ->seeJsonContains([
+                'email' => $user->email,
+            ])
+            ->seeJsonContains([
+                'to' => 3,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_title(): void
+    {
+        QuestionFactory::new([
+            'title' => 'LARAVEL-PT',
+        ])
+            ->create();
+
+        QuestionFactory::new([
+            'title' => 'laravel-Pt',
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'title' => 'LArAvEL-pT',
+        ]))
+            ->seeJsonContains([
+                'to' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_created_date(): void
+    {
+        QuestionFactory::new([
+            'created_at' => Carbon::now()->subYears(2),
+        ])
+            ->create();
+
+        QuestionFactory::new([
+            'created_at' => Carbon::now()->subYears(3),
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->subMonth()->subYears(2)->toDateString(),
+            'created[to]' => Carbon::now()->addMonth()->subYears(2)->toDateString()
+        ]))
+            ->seeJsonContains([
+                'to' => 1,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->subMonth()->subYears(3)->toDateString(),
+            'created[to]' => Carbon::now()->addMonth()->subYears(2)->toDateString()
+        ]))
+            ->seeJsonContains([
+                'to' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_resolved(): void
+    {
+        QuestionFactory::new([
+            'resolved_at' => Carbon::now(),
+        ])
+            ->count(5)
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index'))
+            ->seeJsonContains([
+                'to' => 15,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'resolved' => true,
+        ]))
+            ->seeJsonContains([
+                'to' => 5,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'resolved' => false,
+        ]))
+            ->seeJsonContains([
+                'to' => 10,
+            ]);
+    }
+
+    /**
+     * @test
+     * @dataProvider datesProvider
+     */
+    public function it_fails_by_validations($param, $expected)
+    {
+        $this->get(route('discussions.questions.index', $param))
+            ->assertResponseStatus($expected);;
+    }
+
+    public function datesProvider()
```

Comment:
This data provider name is a bit misleading... It's not only supplying dates, right? What do you think of calling it `invalidSearchablePropertiesValuesProvider`?

Reply:
I understood the question now, I will change it.